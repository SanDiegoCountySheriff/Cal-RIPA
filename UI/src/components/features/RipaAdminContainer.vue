<template>
  <ripa-admin-template
    :loading="loading"
    :beats="beats"
    :cities="cities"
    :schools="schools"
    :statutes="mappedStatutes"
    :on-delete-beat="handleDeleteBeat"
    :on-delete-city="handleDeleteCity"
    :on-delete-school="handleDeleteSchool"
    :on-delete-statute="handleDeleteStatute"
    :on-edit-beat="handleEditBeat"
    :on-edit-city="handleEditCity"
    :on-edit-school="handleEditSchool"
    :on-edit-statute="handleEditStatute"
  ></ripa-admin-template>
</template>

<script>
import RipaAdminTemplate from '@/components/templates/RipaAdminTemplate'
import { mapState, mapGetters, mapActions } from 'vuex'

export default {
  name: 'ripa-admin-container',

  components: {
    RipaAdminTemplate,
  },

  data() {
    return {
      loading: false,
    }
  },

  computed: {
    ...mapState(['beats', 'cities', 'schools', 'isAdmin']),
    ...mapGetters(['mappedStatutes']),
  },

  methods: {
    ...mapActions([
      'deleteBeat',
      'deleteCity',
      'deleteSchool',
      'deleteStatute',
      'editBeat',
      'editCity',
      'editSchool',
      'editStatute',
      'getBeats',
      'getCities',
      'getSchools',
      'getStatutes',
    ]),

    async getAdminData() {
      this.loading = true
      await Promise.all([
        this.getBeats(),
        this.getCities(),
        this.getSchools(),
        this.getStatutes(),
      ])
      this.loading = false
    },

    async handleDeleteBeat(beat) {
      this.loading = true
      await Promise.all([this.deleteBeat(beat)])
      this.loading = false
    },

    async handleDeleteCity(city) {
      this.loading = true
      await Promise.all([this.deleteCity(city)])
      this.loading = false
    },

    async handleDeleteSchool(school) {
      this.loading = true
      await Promise.all([this.deleteSchool(school)])
      this.loading = false
    },

    async handleDeleteStatute(statute) {
      this.loading = true
      await Promise.all([this.deleteStatute(statute)])
      this.loading = false
    },

    async handleEditBeat(beat) {
      this.loading = true
      await Promise.all([this.editBeat(beat)])
      this.loading = false
    },

    async handleEditCity(city) {
      this.loading = true
      await Promise.all([this.editCity(city)])
      this.loading = false
    },

    async handleEditSchool(school) {
      this.loading = true
      await Promise.all([this.editSchool(school)])
      this.loading = false
    },

    async handleEditStatute(statute) {
      this.loading = true
      await Promise.all([this.editStatute(statute)])
      this.loading = false
    },
  },

  created() {
    this.getAdminData()
  },
}
</script>
