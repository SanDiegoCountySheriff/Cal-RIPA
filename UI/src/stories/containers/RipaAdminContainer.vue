<template>
  <ripa-page-container :admin="admin">
    <ripa-admin-template
      :loading="loading"
      :beats="beats"
      :cities="cities"
      :schools="schools"
      :statutes="statutes"
      :stops="stops"
      :submissions="submissions"
    ></ripa-admin-template>
  </ripa-page-container>
</template>

<script>
import RipaPageContainer from './RipaPageContainer'
import RipaAdminTemplate from '@/components/templates/RipaAdminTemplate'
import { beats } from '../data/beats'
import { cities } from '../data/cities'
import { schools } from '../data/schools'
import { statutes } from '../data/statutes'
import { stops } from '../data/stops'
import { submissions } from '../data/submissions'

export default {
  name: 'ripa-admin-container',

  components: {
    RipaPageContainer,
    RipaAdminTemplate,
  },

  data() {
    return {
      loading: false,
      beats: [],
      cities: [],
      schools: [],
      statutes: [],
      stops: [],
      submissions: [],
    }
  },

  methods: {
    getAdminData() {
      this.loading = true
      setTimeout(() => {
        this.beats = beats
        this.cities = cities
        this.schools = schools
        this.statutes = statutes
        this.stops = stops
        this.submissions = submissions
        this.loading = false
      }, 2500)
    },
  },

  created() {
    this.getAdminData()
  },

  props: {
    admin: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
