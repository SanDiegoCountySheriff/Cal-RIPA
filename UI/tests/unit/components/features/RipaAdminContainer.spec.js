import RipaAdminContainer from '@/components/features/RipaAdminContainer'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import Vuetify from 'vuetify'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)

describe('Ripa Admin Container', () => {
  let wrapper
  let vuetify
  let actions
  let getters
  let store
  const routes = [
    {
      path: '/',
      component: RipaAdminContainer,
    },
  ]
  const router = new VueRouter({ routes })

  beforeEach(() => {
    vuetify = new Vuetify()
    actions = {
      getAdminStops: jest.fn(),
    }
    getters = {
      mappedAdminBeats: jest.fn(),
      mappedUser: jest.fn(),
      mappedAdminCities: jest.fn(),
      mappedAdminSchools: jest.fn(),
      mappedAdminStatutes: jest.fn(),
      mappedAdminStops: jest.fn(),
      mappedAdminSubmissions: jest.fn(),
      mappedAdminSubmission: jest.fn(),
      mappedAdminUsers: jest.fn(),
      mappedErrorCodeAdminSearch: jest.fn(),
      displayBeatInput: jest.fn(),
      mappedAdminHistoricalCpraReports: jest.fn(),
      mappedAdminCpraReportStats: jest.fn(),
      stopQueryData: jest.fn(),
    }
    store = new Vuex.Store({
      actions,
      getters,
    })
  })

  const factory = propsData => {
    return shallowMount(RipaAdminContainer, {
      router,
      store,
      localVue,
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = factory()

    expect(wrapper.element).toMatchSnapshot()
  })
})
