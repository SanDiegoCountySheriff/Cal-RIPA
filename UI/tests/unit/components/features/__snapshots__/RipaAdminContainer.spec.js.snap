// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Admin Container should match snapshot 1`] = `
<div>
  <ripa-admin-template-stub beats="" cities="" schools="" statutes="" piientities="" users="" historicalcprareports="" ondeletebeat="[Function]" oneditbeat="[Function]" onedituser="[Function]" onuploadusers="[Function]" ontabchange="[Function]" onuploaddomain="[Function]"></ripa-admin-template-stub>
  <ripa-snackbar-stub timeout="6000" autoclose="true"></ripa-snackbar-stub>
</div>
`;
